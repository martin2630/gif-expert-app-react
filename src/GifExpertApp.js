import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    // const categories = ["one punch", "pokemon", "christmans"];
    const initialCategories = ["one punch"]
    const [categories, setCategories] = useState(initialCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr></hr>
            <div>
                {
                    categories.map( category => (
                            <GifGrid 
                                key={category}
                                category={category}
                            />
                        )
                    )
                }
            </div>
        </>
    )
}
