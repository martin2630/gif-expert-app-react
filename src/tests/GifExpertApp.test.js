import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas GifExpertApp', () => {
    let wrapper = shallow(<GifExpertApp />);
    beforeEach( () => {
            wrapper = shallow(
                <GifExpertApp  />
            );
        }
    )
    test('debe de mostrar <GifExpertApp /> correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    // test('CounterApp debe de mostrar el valor defecto de 100', () => {
    //     const value = 100;
    //     const wrapper = shallow(
    //         <CounterApp 
    //             value={value}
    //         />);
    //     const valueApp = wrapper.find('h2').text().trim();
    //     expect(valueApp).toBe('100');
        
    // })

    // test('debe de incrementar con el boton +1', () => {
    //     wrapper.find("button").at(0).simulate("click");
    //     const valueApp = wrapper.find("h2").text().trim();
    //     expect(valueApp).toBe('11');
        
    // })

    // test('debe de disminuir con el boton -1', () => {
    //     wrapper.find("button").at(2).simulate("click");
    //     const valueApp = wrapper.find("h2").text().trim();
    //     expect(valueApp).toBe('9');
        
    // })


    // test('debe de resetear el valor de counter al valor inicial', () => {
    //     wrapper.find("button").at(1).simulate("click");
    //     const valueApp = wrapper.find("h2").text().trim();
    //     expect(valueApp).toBe('10');
        
    // })
    
    
    
})
