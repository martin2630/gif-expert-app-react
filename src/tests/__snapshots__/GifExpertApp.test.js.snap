// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp debe de mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      category="one punch"
      key="one punch"
    />
  </div>
</Fragment>
`;
